### PR TITLE
Add capability to add SuggestBoxStore after creation of SuggestBox

### PR DESCRIPTION
**Pull request by TDesjardins**
This is especially for using the SuggestBox in a clean MVP manner and
maybe it is also usefull when using templates.

### CODE REVIEW

File: src/main/java/org/dominokit/domino/ui/forms/SuggestBox.java [hunk 1]
```diff
@@ -64,15 +72,17 @@ private void search() {
         loader.start();
```
Comment (vegegoku) [inline src/main/java/org/dominokit/domino/ui/forms/SuggestBox.java]: can we move this into the null check so the loader wont start unless there is a store.

File: src/main/java/org/dominokit/domino/ui/forms/SuggestBox.java [hunk 2]
```diff
@@ -107,6 +117,10 @@ public String getValue() {
         return getInputElement().asElement().value;
     }
 
+    public void setSuggestBoxStore(SuggestBoxStore store) {
```
Comment (vegegoku) [inline src/main/java/org/dominokit/domino/ui/forms/SuggestBox.java]: can we also have a builder style method for this, lets keep the set and make a new method that calls the setter, something like `public SuggestBox suggestStore(SuggetStore)`

File: src/main/java/org/dominokit/domino/ui/forms/SuggestBox.java (hunk 1 above)
Comment (TDesjardins) [reply src/main/java/org/dominokit/domino/ui/forms/SuggestBox.java]: Yep, makes sense!

File: src/main/java/org/dominokit/domino/ui/forms/SuggestBox.java (hunk 2 above)
Comment (TDesjardins) [reply src/main/java/org/dominokit/domino/ui/forms/SuggestBox.java]: There are already two setters (`setLoaderEffect()` and `setType()`) with return type `SuggestBox`. To be consequent I would prefer to fit the setter `setSuggestBoxStore` also with return type `SuggestBox`. WDYT?

Comment (vegegoku) [reply src/main/java/org/dominokit/domino/ui/forms/SuggestBox.java]: go ahead. i agree

**Pull request merged**